v1/keys.go

```diff
@@ -97,3 +102,114 @@ func (r UnwrapRequest) Send(ctx context.Context) (*UnwrapResponse, error) {
 type UnwrapResponse struct {
 	*UnwrapOutput
 }
+
+type CreatePoliciesInput struct {
+	Instance string `json:"-" location:"header" locationKey:"Bluemix-Instance"`
+	KeyID    string `json:"-" location:"url" locationKey:"KeyID"`
+
+	Metadata PoliciesMetadata `json:"metadata"`
+	Policies *[]PolicyDetails `json:"resources,omitempty" mapstructure:"resources"`
+}
+
+type RotationPolicy struct {
+	Interval int `json:"interval_month,omitempty"`
+}
+
+type DualAuthPolicy struct {
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+type PolicyDetails struct {
+	ID        string          `json:"id,omitempty"`
+	Type      string          `json:"type,omitempty"`
+	CreatedBy string          `json:"createdBy,omitempty"`
+	CreatedAt *time.Time      `json:"creationDate,omitempty"`
+	CRN       string          `json:"crn,omitempty"`
+	UpdatedAt *time.Time      `json:"lastUpdateDate,omitempty"`
+	UpdatedBy string          `json:"updatedBy,omitempty"`
+	Rotation  *RotationPolicy `json:"rotation,omitempty" mapstructure:"rotation"`
+	DualAuth  *DualAuthPolicy `json:"dualAuthDelete,omitempty" mapstructure:"dualAuthDelete"`
+}
+
+// PoliciesMetadata represents the metadata of a collection of keys.
+type PoliciesMetadata struct {
+	CollectionType   string `json:"collectionType"`
+	NumberOfPolicies int    `json:"collectionTotal"`
+}
+
+type PoliciesOutput struct {
+	Metadata PoliciesMetadata `json:"metadata"`
+	Policies *[]PolicyDetails `json:"resources"`
+}
+
+type CreatePoliciesRequest struct {
+	*Request
+	Input *CreatePoliciesInput
+}
+
+func (c *Client) CreatePoliciesRequest(in *CreatePoliciesInput) CreatePoliciesRequest {
+	req := NewRequest(c, "PUT", "/api/v2/keys/{{.KeyID}}/policies", in, &PoliciesOutput{})
+
+	return CreatePoliciesRequest{Request: req, Input: in}
+}
+
+func (c *Client) CreateRotationPolicyRequest(in *CreatePoliciesInput) CreatePoliciesRequest {
```

Comment:
Could `CreateRotationPolicyRequest` and `CreateDualAuthPolicyRequest` be handled in `CreatePoliciesRequest`, instead of being separated into functions?  It seems redundant to me to have these functions taking in similar inputs but making requests to different paths. would it be more or less convenient for users? let me know what you think, I am still fairly new to Go.

Reply:
yes , both rotation and dualauth can be handled by `CreatePoliciesRequest`.
I added these methods to support the url with query param.
`CreateRotationPolicyRequest` and `CreateDualAuthPolicyRequest` are not quite  necessary. Just added them to have separate examples like it is currently for curl request in [apidocs](https://test.cloud.ibm.com/apidocs/key-protect#set-key-policies)

Reply:
I think its best to try to keep the number of functions lower, and have a general CreatePolicyRequest